cluster-autoscaler/core/scale_down.go

```diff
@@ -208,6 +216,11 @@ func (sd *ScaleDown) TryToScaleDown(nodes []*apiv1.Node, pods []*apiv1.Pod, pdbs
 			ready, _, _ := kube_util.GetReadinessState(node)
 			readinessMap[node.Name] = ready
 
+			// Check if node is marked with no scale down annotatation
```

Comment:
Moved this above computing node readiness. Is it necessary to move it out of the loop? I think it may be useful to log the reason node wasn't scaled down, but it only makes sense if it was considered unneeded.